services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java

```diff
@@ -328,4 +325,35 @@ public void close() {
   public void addInstanceTag(String clusterName, String instanceName, String tag) {
     helixAdmin.addInstanceTag(clusterName, instanceName, tag);
   }
+
+  public void setCloudConfig(VeniceControllerClusterConfig config, boolean isControllerCluster) {
+    if (isControllerCluster && !config.isControllerClusterHelixCloudEnabled()) {
+      return;
+    } else if (!isControllerCluster && !config.isControllerStorageClusterHelixCloudEnabled()) {
+      return;
+    }
+
+    String controllerCloudProvider = config.getControllerHelixCloudProvider().toUpperCase();
+    CloudProvider cloudProvider;
+    try {
+      cloudProvider = CloudProvider.valueOf(controllerCloudProvider);
+    } catch (IllegalArgumentException e) {
+      throw new VeniceException(
+          "Invalid Helix cloud provider: " + controllerCloudProvider + ". Must be one of: "
+              + Arrays.toString(CloudProvider.values()));
+    }
+
+    String controllerCloudId = config.getControllerHelixCloudId();
+    List<String> controllerCloudInfoSources = config.getControllerHelixCloudInfoSources();
+    String controllerCloudInfoProcessorName = config.getControllerHelixCloudInfoProcessorName();
+    CloudConfig.Builder cloudConfigBuilder =
+        new CloudConfig.Builder().setCloudEnabled(true).setCloudProvider(cloudProvider).setCloudID(controllerCloudId);
```

Comment:
`CLOUD_ID` is an optional field. Let's not set it if the configs don't specify a value for it